apm-lambda-extension/main.go

```diff
@@ -191,6 +197,11 @@ func main() {
 			backgroundDataSendWg.Wait()
 			if config.SendStrategy == extension.SyncFlush {
 				// Flush APM data now that the function invocation has completed
+				for {
+					if extension.CheckTransportStatus() {
+						break
+					}
+				}
 				extension.FlushAPMData(client, agentDataChannel, config)
```

Comment:
Looks like this would block the Lambda runtime until the transport status is healthy again. If APM Server is down for a long period of time, this increased the billed time for all Lambda invocations and avoids the runtimes from being re-used, leading to lots of cold starts.
It also busy-spins, saturating a CPU core.

Instead, can we just flush like so:

```suggestion
				extension.FlushAPMData(client, agentDataChannel, config)
```

And exit early in

https://github.com/elastic/apm-aws-lambda/blob/8e471917dcb9e05b473243f99dcadd01a89750ea/apm-lambda-extension/extension/process_events.go#L32-L34

```go
if !IsTransportStatusHealthy() {
	return
}
```